tests/unittests/test_creators/test_keypoints.py

```diff
@@ -1,155 +1,133 @@
-        ),
+        ValueError, match="The confidence_threshold should be float, got <class 'str'>."
     ):
-        create_keypoints_message(
-            keypoints=[[1.0, 2.0, 3.0], [1.0, 3.0], [1.0, 2.0, 3.0]]
-        )
+        create_keypoints_message(keypoints, scores, confidence_threshold="not a float")
 
 
-def test_nonlist_keypoint():
+def test_confidence_threshold_out_of_range():
+    keypoints = [[0.1, 0.2], [0.3, 0.4]]
+    scores = [0.9, 0.8]
     with pytest.raises(
-        ValueError,
-        match="Keypoints should be list of lists or np.array, got list of <class 'str'>.",
+        ValueError, match="The confidence_threshold should be between 0 and 1."
     ):
-        create_keypoints_message(keypoints=[[1.0, 2.0, 3.0], "1.0"])
+        create_keypoints_message(keypoints, scores, confidence_threshold=1.1)
 
 
-def test_nonfloat_keypoint():
+def test_invalid_keypoints_shape():
+    keypoints = [[0.1, 0.2, 0.3, 0.4]]
     with pytest.raises(
         ValueError,
-        match="Keypoints inner list should contain only float, got <class 'str'>.",
+        match="All keypoints should be of dimension 2 or 3, got dimension 4.",
     ):
-        create_keypoints_message(keypoints=[["1.0", 2.0, 3.0]])
+        create_keypoints_message(keypoints)
 
 
-def test_keypoints_shape():
+def test_invalid_keypoints_inner_type():
+    keypoints = [[0.1, "not a float"]]
     with pytest.raises(
         ValueError,
-        match=re.escape("Keypoints should be of shape (N,2 or 3) got (3, 2, 3)."),
+        match="Keypoints inner list should contain only float, got <class 'str'>.",
     ):
-        create_keypoints_message(
-            np.array(
-                [
-                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
-                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
-                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
-                ]
-            )
-        )
-
-
-def test_all_scores_below_thr():
-    message = create_keypoints_message([[1.0, 2.0, 3.0]], [0.4], 0.6)
-
-    assert isinstance(message, Keypoints)
-    assert message.keypoints == []
-
-
-def test_all_scores_above_thr():
-    message = create_keypoints_message([[0.1, 0.2, 0.3]], [0.8], 0.6)
-
-    true_pt = np.array([0.1, 0.2, 0.3])
-
-    assert len(message.keypoints) == 1
-    assert isinstance(message, Keypoints)
-    assert isinstance(message.keypoints[0], Keypoint)
-    assert message.keypoints[0].x == true_pt[0]
-    assert message.keypoints[0].y == true_pt[1]
-    assert message.keypoints[0].z == true_pt[2]
-
-
-def test_2d_keypoints():
-    message = create_keypoints_message([[0.1, 0.2], [0.3, 0.4]], [0.8, 0.5], 0.6)
-
-    assert len(message.keypoints) == 1
-    assert message.keypoints[0].x == 0.1
-    assert message.keypoints[0].y == 0.2
-    assert message.keypoints[0].z == 0.0
-
-
-if __name__ == "__main__":
-    pytest.main()
+        create_keypoints_message(keypoints)
```

Comment:
We are missing a test to check that all keypoints are of the same dimension e.g. [x, y] or [x, y, z]

Reply:
True, thanks. Added that check.